fastlane/lanes/localization.rb

```diff
@@ -127,55 +129,58 @@
+    # That's because we haven't yet implemented a method to share the derived data folder explicitly between the CI SPM caching logic and this lane.
+    #
+    # See also:
+    # - https://github.com/wordpress-mobile/WordPress-iOS/pull/23506/files/f694d2c3f433d616deeb15a61a3e9a01f55bc07e#r1725934262
+    # - https://github.com/wordpress-mobile/WordPress-iOS/pull/23506#discussion_r1727419190
+    derived_data_path = options.fetch(:derived_data_path, DERIVED_DATA_PATH)
+    resolve_packages(derived_data_path: derived_data_path)
+
+    custom_gutenberg_path = options[:gutenberg_path]
+    if custom_gutenberg_path
+      # It's simpler to ask the caller to give an absolute path than to make it absolute ourselves
+      # Given this is a debug option, it seems like an okay tradeoff.
+      UI.user_error!("Please provide gutenberg_path as an absolute path. Current value: #{custom_gutenberg_path}") unless File.absolute_path?(custom_gutenberg_path)
+
+      UI.user_error!("Could not find Gutenber project at given path #{custom_gutenberg_path}") unless Dir.exist?(custom_gutenberg_path)
+
+      UI.message("Using Gutenberg from #{custom_gutenberg_path} instead of cloning it...")
+      generate_strings_file(gutenberg_path: custom_gutenberg_path, derived_data_path: derived_data_path)
+    else
+      # On top of fetching the latest Pods, we also need to fetch the source for the Gutenberg code.
+      # To get it, we need to manually clone the repo, since Gutenberg is distributed via XCFramework.
+      # XCFrameworks are binary targets and cannot extract strings via genstrings from there.
+      config = gutenberg_config!
+
+      ref_node = config[:ref]
+      UI.user_error!('Could not find Gutenberg ref to clone the repository in order to access its strings.') if ref_node.nil?
+
+      ref = ref_node[:tag] || ref_node[:commit]
+      UI.user_error!('The ref to clone Gutenberg in order to access its strings has neither tag nor commit values.') if ref.nil?
+
+      github_org = config[:github_org]
+      UI.user_error!('Could not find GitHub organization name to clone Gutenberg in order to access its strings.') if github_org.nil?
+
+      repo_name = config[:repo_name]
+      UI.user_error!('Could not find GitHub repository name to clone Gutenberg in order to access its strings.') if repo_name.nil?
+
+      # Create a temporary directory to clone Gutenberg into.
+      Dir.mktmpdir do |tempdir|
         gutenberg_clone_name = 'Gutenberg-Strings-Clone'
+        gutenberg_path = File.join(tempdir, gutenberg_clone_name)
         Dir.chdir(tempdir) do
           repo_url = "https://github.com/#{github_org}/#{repo_name}"
           UI.message("Cloning Gutenberg from #{repo_url} into #{gutenberg_clone_name}. This might take a few minutes…")
-          sh("git clone --depth 1 #{repo_url} #{gutenberg_clone_name}")
-          Dir.chdir(gutenberg_clone_name) do
-            if config[:ref][:tag]
+          sh("git clone --depth 1 #{repo_url} #{gutenberg_path}")
```

Comment:
It took me so long to understand this comment because in my mind there always was just one `Dir.chdir` in the whole code... 😅 Yes, totally. Not sure how I ended up with _two_ 🤦‍♂️